src/index.js

```diff
@@ -72,32 +72,34 @@ module.exports = class FederatedTypesPlugin {
   }
 
   importRemoteTypes() {
-    const remoteUrls = Object.values(this.remoteComponents).map((r) => {
+    const remoteUrls = Object.values(this.remoteComponents).map(r => {
+      let remote;
       let url;
-
-      if (r.includes("@")) {
-        url = r.split("@")[1];
+    
+      if (r.includes('@')) {
+        remote = r.split('@')[0];
+        url = r.split('@')[1];
       }
-
+    
       const resolvedURL = new URL(url);
-      return resolvedURL.origin;
+    
+      return { 
+        origin: resolvedURL.origin,
+        remote
```

Comment:
Remote can't always have a value, In the event the remote is defined as `http://..` instead of `app@http://...`.